src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -96,7 +96,6 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-Wl,--as-needed" Condition="'$(TargetOS)' != 'OSX'" />
       <LinkerArg Include="-Wl,-e0x0" Condition="'$(NativeLib)' == 'Shared' and '$(TargetOS)' != 'OSX'" />
       <LinkerArg Include="-pthread" Condition="'$(TargetOS)' != 'OSX'" />
-      <LinkerArg Include="-lstdc++" />
```

Comment:
Will this cause a local build to fail or only the testing? I'm concerned about how easy it is to introduce a C++ dependency. By doing this we will be imposing the restriction on all aspects of native code correct? I'd assume this would have knock-on effects in the GC as well. If not, we will need some way to ensure they aren't introduced and then need to be rewritten.

Reply:
Full `./build.sh` or `packs` subset in particular will fail to generate native code for crossgen2 binary.

Reply:
This is something we should discuss for sure. Compiling cpp files with the expectation they aren't using the C++ runtime but on what is acceptable is rather annoying. I would personally prefer if we just compiled them as C if that is what we want. However, this is a much broader conversation. I think your current pass makes sense, but we need a real policy in place that we all agree on. Give me some time, but for this particular PR I think I am fine for now as long as @jkotas and others are on board.

Reply:
> I have "replace new's with malloc's" bulk change (wip) stashed if needed. :)

Could you please share it? I would like to see what it looks like.

Reply:
https://github.com/dotnet/runtime/compare/main...am11:runtime:feature/nativeaot/convert-new_delete-to-malloc_free

This was wip when I decided to use the shim approach and stashed it. The ctor calls after malloc+nullcheck can be converted to `initialize()` pattern in pure C.

Reply:
> > I have "replace new's with malloc's" bulk change (wip) stashed if needed. :)

> Could you please share it? I would like to see what it looks like.

I'm actually far less concerned about these changes.

> The ctor calls after malloc+nullcheck can be converted to initialize() pattern in pure C.

If you updated that diff with this, I'd be fine moving forward with that PR. @Maoni0 Thoughts on making more gc code more C-ish when possible?

Reply:
> I'm actually far less concerned about these changes.

Yes, I would prefer something like this too.